### DISCUSSION TITLE
Should tB Run EXE After Build?

### DISCUSSION THREAD

**Discussion by jpbro** [twinbasic/twinbasic#615] [Q&A] [answered]
I'm plugging away the old-fashioned way on the memory leak issue discussed in #614 by copying chunks of code to a new project until I can determine the source of the problem. I was a bit confused at first because I would run my program after building and then see 2 processes in the Task Manager. I eventually realized that tB runs the EXE immediately after building, and I found this a bit unintuitive. Would it be better to have separate _Build_ and _Build & Run_  buttons?

**Comment by WaynePhillipsEA** [comment, marked as answer]
In the project Settings, there is an option to turn this off:

<img width="577" alt="tbRunAfterBuild" src="https://user-images.githubusercontent.com/25738109/147731036-f753d482-6c84-429f-9029-0ce17c1f0a55.png">

However, I quite like the idea of having two buttons.

**Reply by EduardoVB** [reply]
FWIW, I second the idea of having two buttons.
When I build a real program, I rarely want to run it (immediately).

**Reply by WaynePhillipsEA** [reply]
I'll gladly add a second button, just need to decide on an icon for it.  Ideas welcome.

**Reply by Kr00l** [reply]
What happens with the project setting when two buttons are available?

**Reply by WaynePhillipsEA** [reply]
I guess it will no longer be needed

**Reply by Kr00l** [reply]
> I'll gladly add a second button, just need to decide on an icon for it. Ideas welcome.

Build icon with a running man. (?)

**Reply by Kr00l** [reply]
> I guess it will no longer be needed

That's good. I would also ensure that this "build and run" icon is visible only for EXE projects.
